skbio/stats/composition.py

```diff
@@ -608,10 +611,329 @@ def centralize(mat):
+
+    if multiple_comparisons_correction is not None:
+        if multiple_comparisons_correction != 'holm-bonferroni':
+            raise ValueError('%s is not an option for '
+                             '`multiple_comparisons_correction`'
+                             % multiple_comparisons_correction)
+
+    if significance_test is None:
+        significance_test = scipy.stats.f_oneway
+
+    mat = _check_composition(table, ignore_zeros=False)
+    cats = pd.Series(grouping)
+
+    mat.sort_index(inplace=True)
+    cats = cats.sort_index()
+    labs = mat.columns
+
+    n_samp, n_feat = mat.shape
+
+    _logratio_mat = _log_compare(mat.values, cats.values, significance_test)
+    logratio_mat = _logratio_mat + _logratio_mat.T
+
+    # Multiple comparisons
+    if multiple_comparisons_correction == 'holm-bonferroni':
+        logratio_mat = np.apply_along_axis(_holm_bonferroni,
+                                           1, logratio_mat)
+    np.fill_diagonal(logratio_mat, 1)
+    W = (logratio_mat < alpha).sum(axis=1)
+    c_start = W.max() / n_feat
+    if c_start < theta:
+        reject = np.zeros_like(W, dtype=bool)
+    else:
+        # Select appropriate cutoff
+        cutoff = c_start - np.linspace(0.05, 0.25, 5)
+        prop_cut = np.array([(W > n_feat*cut).mean() for cut in cutoff])
+        dels = np.abs(prop_cut - np.roll(prop_cut, -1))
+        dels[-1] = 0
+        if (dels[1] < tau) and (dels[2] < tau) and (dels[3] < tau):
+            nu = cutoff[1]
+        elif (dels[1] >= tau) and (dels[2] < tau) and (dels[3] < tau):
+            nu = cutoff[2]
+        elif (dels[2] >= tau) and (dels[3] < tau) and (dels[4] < tau):
+            nu = cutoff[3]
+        else:
+            nu = cutoff[4]
+        reject = (W >= nu*n_feat)
+    return pd.DataFrame({'W': pd.Series(W, index=labs),
+                         'reject': pd.Series(reject, index=labs)})
+
+
+def _check_composition(x, ignore_zeros=True):
+    """ Checks to make sure that composition meets the mininum criteria
+
+    Also casts composition into a pandas dataframe
+
+    Parameters
+    ----------
+    x : array_like or pd.DataFrame
+       Input composition matrix
+       where rows=samples and columns=features
+    ignore_zeros : bool
+       Specifies if compositions are allowed to have zeros
+
+    Returns
+    -------
+    pd.DataFrame
+       Validated composition matrix
+    """
+    if x.ndim > 2:
+        raise ValueError("Input matrix can onlx have two dimensions or less")
+    if isinstance(x, pd.DataFrame):
+        mat = closure(x)
+        samp_labs = x.index
+        feat_labs = x.columns
+    else:
+        mat = np.atleast_2d(x)
+        r, c = mat.shape
+        samp_labs = np.arange(r)
+        feat_labs = np.arange(c)
+    if np.any(mat == 0) and not ignore_zeros:
+        raise ValueError('Cannot handle zeros in compositions. '
+                         'Make sure to run a multiplicative_replacement')
```

Comment:
Update error text to be:

    Cannot handle zeros in compositions. Use pseudo counts or `multiplicative_replacement`.

That way the error message's recommendation matches the docstring.